pkg/utils/utils.go

```diff
@@ -197,6 +199,22 @@ func GetS3PETMFromParamsMap(params map[string]interface{}, logger logrus.FieldLo
 		Region: aws.String(region),
 	}))
 
+	s3Url, ok := params["s3Url"].(string)
+	if ok {
+		sess.Config.Endpoint = aws.String(s3Url)
+	}
+
+	pathStyle, ok := params["s3ForcePathStyle"].(string)
+	var s3ForcePathStyle bool
+	if pathStyle == "true" {
```

Comment:
This code is a little unclear
If the code was this:
`
pathStyle, ok := params["s3ForcePathStyle"]
`
ok is true if s3ForcePathStyle was found in the map

However, we have:
`  pathStyle, ok := params["s3ForcePathStyle"].(string)
`
ok is true if the value exists and is a string, which avoids a panic if the value is not a string (good) but doesn't say anything if it's not a string which can make debugging more difficult. 

The code as it is written, checks pathStyle even if ok is false which is a little confusing.  We only set sess.Config.S3ForcePathStyle if ok is true, but we set s3ForcePathStyle regardless of ok.  If there's a non-string value in the map we will silently ignore it.

Since we have multiple spots where we need to get the value as a string let's add a function

```
func getStringParam(paramMap map[string]interface{}, key string, logger logrus.FieldLogger) (value string, ok bool) {
	valueIF, ok := paramMap[key]
	if ok {
		value, ok = valueIF.(string)
	} else {
            logger.Errorf("param key %s is not a string", key)
        }
	return
}

```
Then the code should look more like this:

```
pathStyle, ok := getStringParam(params, "s3ForcePathStyle", logger)

if ok {
     if GetBool(pathStyle, false) {
	   sess.Config.S3ForcePathStyle = aws.Bool(true)
           logger.Info("Setting s3ForcePathStyle to true")
     } else {
           logger.Infof("Got %s for s3ForcePathStyle, setting s3ForcePathStyle to false", pathStyle)
    }
}
```
We assume sess.Config.S3ForcePathStyle is false so we don't set it if not true.  The logging is so when the user sets s3ForcePathStyle to "ture" and then complains that we're not doing the right thing we don't spend days trying to figure out what was actually in their configuration.  We will also get an error message if they manage to put a non-string value into the map.

Let's make the other param usages similar.

Reply:
Added a GetStringFromParamMap method to provide more useful log message.